### PR TITLE
Improvements to development docker-compose setup

### PR DESCRIPTION
**Pull request by bcongdon**
- Now using volumes to live-reload development code from projects that support this
   - Some services like the api-gateway won't work like this, you'll have to still rebuild the container because the gateway needs to be recompiled
- Set the `-u` flag on api-gateway to not require gateway authentication while doing dev work
- Set debug environment variables to allow python services to live-reload

This can be merged as-is, but if we want full live-reload support we'll need:
- A way to live-reload node services -- maybe [nodemon](https://www.npmjs.com/package/nodemon)?
- A way to live-reload ruby services -- I think Sinatra supports this, we just have to set a development flag somewhere

Additionally, we'll need to patch the Dockerfiles for Ruby services to move `wait-for-it` out of the application source directory. See https://github.com/acm-uiuc/groot-users-service/pull/15 for an example

Ruby service dockerfiles needing patch:
- [x] groot-users-service
- [x] groot-auth-stub-service
- [x] groot-merch-service
- [x] groot-quotes-service
- [x] groot-recruiters-service

### CODE REVIEW

**Comment by bcongdon** [comment]
@sameetandpotatoes: I think we can use `Sinatra::Reloader` for our ruby services? http://www.sinatrarb.com/contrib/reloader

**Comment by narendasan** [comment]
how do we deploy this (like removing the -u flag)?
is it possible to create a hook or something to trigger a script in a container (for rebuilding the gateway) or should we just kill that container and restart it?
If containers are killed can we spin them backup automatically?
In this PR or a separate one can we also add the unified env file as well?

**Comment by bcongdon** [comment]
@narendasan:
1) We have a separate [prod docker-compose](https://github.com/acm-uiuc/groot/blob/master/docker-compose-prod.yml) file
2) This isn't possible within docker-compose, but there are [other solutions](http://blog.fgribreau.com/2017/04/docker-compose-watch-k-docker-compose.html) using something like nodemon that could emulate this. I'd prefer to have services that _can_ reload themselves manage that themselves, as there's less restart overhead from docker that way. I'll look into a livereload solution for the api gateway.
3) It looks like if you `docker kill` a container, it doesn't restart. If the container processes crashes, or otherwise exits, it does restart. However, running `docker-compose up` will start any containers that need to be started.
4) Yeah, that'll take some more work, but that's still on my radar.

**Comment by bcongdon** [comment]
This can be merged as-is. The work to add live-reloading to Ruby / Node / Go services can be added tracked in separate issues:

https://github.com/acm-uiuc/groot/issues/48
https://github.com/acm-uiuc/groot/issues/49
https://github.com/acm-uiuc/groot-api-gateway/issues/69

File: docker-compose.yml [hunk 1]
```diff
@@ -110,6 +137,8 @@ services:
     depends_on:
       - db
       - groot-api-gateway
+    volumes:
+      - ./groot-users-service:/usr/src/app
```
Comment (sameetandpotatoes) [inline docker-compose.yml]: Do we need to add `RACK_ENV=development` to the ruby services? Or is development assumed by default?

Comment (bcongdon) [reply docker-compose.yml]: Development is the default

**Pull request merged**